### ISSUE TITLE
false positive for match_ref_pats on trivial match

### ISSUE THREAD

**Issue by durka** [Manishearth/rust-clippy#335]
I happened to have a macro-generated match that looks something like this:

```rust
match 42 { _ => () }
```

This triggers the `match_ref_pats` lint. Now this is certainly a stupid match, but it doesn't have `&` prefixed to any of the patterns, and the suggestion `match *42 { ... }` is a bad idea, so it should trigger a different lint if any.

Also, the lint may need a macro check since that's how I discovered this.

**Comment by llogiq** [comment]
Good catch! I'll add a macro check, and perhaps also a check for literals.

**Comment by durka** [comment]
It's not a literal that's the problem -- in my original program it was matching on a variable -- it's the fact that there are no patterns except `_`, seems like.
